src/editpage/components/assistance/SettingsModal.vue

```diff
@@ -0,0 +1,232 @@
+<template>
+    <div id="settingsModal" class="modal">
+        <!-- Modal content -->
+        <div id="settingsModalContent">
+          <button id="closeSettingsModalButton" class="settings-modal-button"
+              @click="closeSettingsModal">
+            &times;
+        </button>
+          <!-- <h1>
+            Settings
+          </h1> -->
+          <h3>
+            Current configuration
+          </h3>
+          <p id="settingsOverview">
+          </p>
+          <h3>
+            View options
+          </h3>
+        <!-- <div class="slidecontainer">
+            <input v-model="zoomSliderValue" type="range" min="0.3" max="3"
+            step="0.1"
+            value="50" class="slider" id="myRange" @change="zoomSliderChanged">
+        </div> -->
+        <table class='padding-table-columns'>
+          <tr>
+            <th><h5 style='padding-right: 50px;'>Change the zoom level</h5></th>
+            <th>
+              <button class='settings-modal-button' @click="zoomIn">
+                Zoom in
+              </button>
+              &nbsp;
+              <button class='settings-modal-button' @click="zoomOut">
+                Zoom out
+              </button>
+            </th>
+          </tr>
+          <tr>
+            <th>
+              <h5 style='padding-right: 50px;'>Theme</h5>
+            </th>
+            <th>
+              <div class="dropdown">
+                <button class="dropbtn settings-modal-button">Select</button>
+                <div class="dropdown-content">
+                  <a v-for="style in styles" :key="style"
+                      @click="changeTheme(style)">
+                      {{style}}</a>
+                </div>
+              </div>
+            </th>
+          </tr>
+        </table>
+
+      </div>
+    </div>
+</template>
+
+
+<script>
+import Vue from 'vue';
+
+export default {
+  name: 'SettingsModal',
+  components: {},
+  props: {},
+  data: function() {
+    return {
+      eventBus: new Vue(),
```

Comment:
Within this component I don't think we need an entire eventbus

Reply:
I'm not sure, but is there a better way to enable it from Sidebar.vue in that case? Now, we use that event bus